src/main/java/Tower.java

```diff
@@ -1,48 +1,31 @@
 import java.util.*;
+import java.util.stream.Stream;
 
 public class Tower {
 
-    private List<Athlete> athletes;
+    private Stream<Athlete> athletes;
 
     public Tower(List<Athlete> athletes) {
-        this.athletes = athletes;
+        this.athletes = sortAthletesByLighterFirst(athletes);
     }
 
-    public Integer getMaxAthletesInPyramid() {
-        orderAthletesByMass();
-
-        return calculateMaxAthletes();
-    }
-
-    private void orderAthletesByMass() {
-        athletes.sort(Comparator.comparing(Athlete::getMass).reversed());
+    private Stream<Athlete> sortAthletesByLighterFirst(List<Athlete> athletes) {
+        return athletes
+                .stream()
+                .sorted();
     }
 
-    private Integer calculateMaxAthletes() {
-        int sumOfLighterAthletes = 0;
-        int sum;
-        int maxAthletes = 1;
-        int max = 1;
+    public int getMaxAthletes() {
+        List<Athlete> athletesInTower = new ArrayList<>();
 
-        for (int i = 0; i < athletes.size() - 1; i++) {
-            for (int j = i + 1; j < athletes.size(); j++) {
-                sum = sumOfLighterAthletes + athletes.get(j).getMass();
-                if (sum <= athletes.get(i).getStrength()) {
-                    sumOfLighterAthletes = sum;
-
-                    maxAthletes = updateMaxAthletes(maxAthletes, ++max);
-                } else {
-                    break;
-                }
+        athletes.forEach(athlete -> {
+            if (athlete.canHold(athletesInTower)) {
+                athletesInTower.add(athlete);
+            } else {
+                return;
             }
-            max = 1;
-            sumOfLighterAthletes = 0;
-        }
-
-        return maxAthletes;
-    }
+        });
 
-    private int updateMaxAthletes(int maxAthletes, int max) {
-        return Math.max(maxAthletes, max);
+        return athletesInTower.size();
```

Comment:
Eu achei a lógica anterior meio complicada, com 2 loops, variáveis temporárias, somas, etc. Tentei implementar algo que reproduzisse a lógica que eu tinha na minha cabeça:
- ordenar os atletas do mais leve pro mais pesado
- verificar se o atleta da vez aguenta segurar todos os anteriores
- até que chegue algum sem força suficiente 
- e fim do cálculo